project/Dependencies.scala

```diff
@@ -45,7 +45,7 @@ object Dependencies {
   val fs2Core             = "co.fs2"                     %% "fs2-core"                 % "0.10.4"
   val fs2Io               = "co.fs2"                     %% "fs2-io"                   % "0.10.4"
 
-  val sparkCore           = "org.apache.spark"           %% "spark-core"               % Version.spark
+  val sparkCore           = "org.apache.spark"            % "spark-core_2.11"          % Version.spark
```

Comment:
How could this produces a correct artifact for `geotrellis-spark`? Is it just succeeding in compiling something that won't actually run?

Reply:
@echeipesh yea, the motivation of this PR was from the one hand - to check how gt at least compiled against 2.12, from the other to help RF team with 2.12 tests. From this PR we can see that all non sparky geotrellis parts have all tests passed.

Reply:
So this PR allows to publish 2.12 gt artifacts, and all non spark part would work as expected. Dirty but working and tested. My vote would be to backport covariance fixes into master and to wait for spark 2.4 artifacts.